src/library/dtls.cpp

```diff
@@ -80,6 +81,47 @@ static void HandleMbedtlsDebug(void *, int level, const char *file, int line, co
     }
 }
 
+/**
+ * This function convert mbedtls error to OT Commissioner error.
+ *
+ * For the implementation details, please reference to <mbedtls/error.h>.
+ *
+ */
+static Error ErrorFromMbedtlsError(int mbedtlsError)
+{
+    ASSERT(mbedtlsError <= 0);
+
+    Error error;
+
+    uint16_t highLevelModuleId = (static_cast<uint16_t>(-mbedtlsError) >> 12) & 0x07;
+
+    if (mbedtlsError == 0)
+    {
+        error = Error::kNone;
+    }
+    else if (mbedtlsError == MBEDTLS_ERR_SSL_WANT_READ || mbedtlsError == MBEDTLS_ERR_SSL_WANT_WRITE ||
+             mbedtlsError == MBEDTLS_ERR_SSL_ASYNC_IN_PROGRESS || mbedtlsError == MBEDTLS_ERR_SSL_CRYPTO_IN_PROGRESS)
+    {
+        error = Error::kTransportBusy;
+    }
+    else if (mbedtlsError <= -0x0042 && mbedtlsError >= -0x0052)
```

Comment:
is it possible to avoid these magic numbers and those in line 112?

Reply:
I don't think so. There are no macros indicating the begin and end of the error code of specific category, for example `MBEDTLS_SSL_ERROR_BEGIN` and `MBEDTLS_SSL_ERROR_END`. This could be a PR to mbedtls, but at least I can define a local macro for our usage. Will update.